### ISSUE TITLE
molecule doesn't honour --destroy=never

### ISSUE THREAD

**Issue by dennisconrad** [metacloud/molecule#1057]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible 2.4.2.0
  config file = None
  configured module search path = [u'/Users/dennis.conrad/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dennis.conrad/.pyenv/versions/2.7.14/lib/python2.7/site-packages/ansible
  executable location = /Users/dennis.conrad/.pyenv/versions/2.7.14/bin/ansible
  python version = 2.7.14 (default, Nov 20 2017, 11:18:37) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

```
molecule, version 2.5.1.dev2
```

- Molecule installation method: source
- Ansible installation method: pip

# Description of Issue

`--destroy=never` doesn't seem to do what it implies to do.  The only difference between `--destroy=never` and `--destroy=always` is that in the case of failed tests, the instances are not destroyed unless `--destroy=always` has been set.

# Desired Behaviour

This is what I expect:

`--destroy`| lint | destroy | dependency-verify | destroy if succeeded / failed
--- | --- | --- | --- | ---
`never` | ✔️ | ✖️  | ✔️ |  ✖️ / ✖️  
`always` | ✔️ | ✔️  | ✔️ |  ✔️ / ✔️

Basically, _never_ destroy my instances when `--destroy=never` has been requested.

# Actual Behaviour (Bug report only)

This is what is actually happening:

`--destroy`| lint | destroy | dependency-verify | destroy if succeeded / failed
--- | --- | --- | --- | ---
`never` | ✔️ | ✔️  | ✔️ | ✔️ / ✖️  
`always` | ✔️ | ✔️  | ✔️ | ✔️ / ✔️

Even with `--destroy=never`, instances are still _always_ destroyed unless the there is a problem.

**Comment by dennisconrad** [comment]
PR for above: https://github.com/metacloud/molecule/pull/1058

**Comment by retr0h** [comment]
Fixed in #1058.
